projects/plugins/jetpack/modules/masterbar/admin-menu/class-atomic-admin-menu.php

```diff
@@ -277,4 +280,23 @@ public function add_gutenberg_menus( $wp_admin = false ) {
 		remove_menu_page( 'gutenberg' );
 		parent::add_gutenberg_menus( $wp_admin );
 	}
+
+	/**
+	 * Saves the sidebar state ( expanded / collapsed ) via an ajax request.
+	 */
+	public function ajax_sidebar_state() {
+		$expanded = filter_var( $_REQUEST['expanded'], FILTER_VALIDATE_BOOLEAN ); // phpcs:ignore WordPress.Security.NonceVerification.Recommended
+		Client::wpcom_json_api_request_as_user(
+			'/me/preferences',
+			'2',
+			array(
+				'method' => 'POST',
+			),
+			(object) array( 'calypso_preferences' => (object) array( 'sidebarCollapsed' => ! $expanded ) ),
+			'wpcom'
+		);
+
+		die();
```

Comment:
Seems that `wp_die` is preferred: https://codex.wordpress.org/AJAX_in_Plugins#Ajax_on_the_Administration_Side

> Notice the use of wp_die(), instead of die() or exit(). Most of the time you should be using wp_die() in your Ajax callback function. This provides better integration with WordPress and makes it easier to test your code.

Reply:
Good catch! Done